tests/test_cache.json

```diff
@@ -1,6 +1,10 @@
ales Figures by Brand | GCBC', 'link': 'https://www.goodcarbadcar.net/2022-us-vehicle-sales-figures-by-brand/', 'snippet': '2022 US Vehicle Sales Figures By Brand. Comprehensive data tables and charts for automotive sales in the United States. Home.'}, {'title': 'USA - Flash report, Automotive sales volume, 2023 - MarkLines ...', 'link': 'https://www.marklines.com/en/statistics/flash_sales/automotive-sales-in-usa-by-month', 'snippet': 'Light vehicle seasonally adjusted annualized sales (SAAR) for February 2023 was 15.19 million units versus 13.96 million units one year ago. With its vehicle\\xa0...'}, {'title': 'August U.S. auto sales: Toyota, Honda skid again; Ford, Hyundai ...', 'link': 'https://www.autonews.com/sales/august-us-auto-sales-toyota-honda-skid-again-ford-hyundai-kia-post-double-digit-august-gains', 'snippet': 'Sep 1, 2022 ... LMC Automotive on Friday slashed its outlook for 2022 sales to 13.8 million from 14.3 million. S&P Global Mobility last week reduced its\\xa0...'}, {'title': 'NADA Releases Third Quarter 2022 U.S. Auto Sales Analysis | NADA', 'link': 'https://www.nada.org/nada/press-releases/nada-releases-third-quarter-2022-us-auto-sales-analysis', 'snippet': 'Oct 12, 2022 ... The third quarter of 2022 closed with a new light-vehicle sales totaling a quarterly average SAAR of 13.3 million units, down slightly from a\\xa0...'}, {'title': 'Full-Year 2022 National Auto Sales By Brand', 'link': 'https://www.mysanantonio.com/news/article/Full-Year-2022-National-Auto-Sales-By-Brand-17713421.php', 'snippet': 'Jan 12, 2023 ... ... brand for the full year of 2022 and how the numbers fared versus the year of 2021: Brand. 2022 U.S. Sales. VS 2021. 1. Toyota, 1,849,751, Down 9\\xa0...'}, {'title': 'NADA Issues Analysis of 2022 Auto Sales and 2023 Sales Forecast ...', 'link': 'https://www.nada.org/nada/press-releases/nada-issues-analysis-2022-auto-sales-and-2023-sales-forecast', 'snippet': 'Jan 11, 2023 ... In 2023, NADA anticipates new-vehicle sales of 14.6 million units \u2013 an increase of 6.6% from 2022. At the macro level, inflation has begun to\\xa0...'}, {'title': 'Hybrid-Electric, Plug-in Hybrid-Electric and Electric Vehicle Sales ...', 'link': 'https://www.bts.gov/content/gasoline-hybrid-and-electric-vehicle-sales', 'snippet': '21, 2022. Publications: National Transportation Statistics Table 1-19.'}, {'title': 'Market Beat | NADA', 'link': 'https://www.nada.org/nada/research-and-data/market-beat', 'snippet': '... U.S. new light vehicle sales; it replaces the NADA Monthly Sales Recap ... 2022 Archives. December NADA Market Beat: 2022 New Light-Vehicle Sales top\\xa0...'}, {'title': 'Light Duty Electric Drive Vehicles Monthly Sales Updates | Argonne ...', 'link': 'https://www.anl.gov/esia/light-duty-electric-drive-vehicles-monthly-sales-updates', 'snippet': '... United States, up 68.1% from the sales in May 2022. PEVs captured 8.36% of ... Vehicle Emissions,\u201d Argonne National Laboratory, Lemont, IL USA, ANL/ESD-22\\xa0...'}]"}
```

Comment:
@asl3 This PR looks good overall.
Only one question here: why there are so many code diff on the cache json file? Is the branch based on the latest master branch?